GOVERNANCE.md

```diff
@@ -0,0 +1,140 @@
+
+- Overseeing the project health and growth
+- Maintaining the brand, mission, vision, values, and scope of the overall project
+- Changes to licensing and intellectual property
+- Administering access to all project assets
+- Administering git repositories as needed
+- Handling Code of Conduct violations
+- Managing financial decisions
+- Defining the scope of each git repository
+- Resolving escalated decisions when the git repository Maintainers responsible are blocked
+
+Ultimately the committee - after consulting with the collective of Maintainers and their community - drive the direction, values and governance of the overall project.
+
+This committee will initially be comprised of Flux Maintainers who have steered the project prior to this initial Governance document.
+The aspiration is no one company or organization should have oversight of the overall project, however that is not yet realistic at this stage. The goal is to broaden maintainership to include a wider range of organizations during CNCF incubation.
+
+Oversight Committee members are publicly listed the `@fluxcd/oversight-committee` <https://github.com/orgs/fluxcd/teams/oversight-committee> GitHub team.
+
+## Proposal Process
+
+- Code changes should go through the pull request process, where the idea and implementation details can be publicly discussed with Maintainers, other contributors, and end users.
+  Pull requests should only be merged after receiving GitHub approval from at least one other Maintainer.
+  Voting is not required for most code changes.
+  Note that Flux v2 uses GitHub discussions for proposals in the `fluxcd/toolkit` git repository <https://github.com/fluxcd/toolkit/discussions?discussions_q=category%3AProposals>.
+- Non-code changes should be proposed as GitHub issues.
+  If unclear which git repository to create the issue in, default to the community repository <https://github.com/fluxcd/community>.
+
+## Decision Making
+
+- Decisions that affect only one git repository may be made informally by its Maintainers.
+- Decisions that affect multiple repositories, or otherwise warrant wider input, should go through the proposal process.
+- All proposals should be discussed publicly in an appropriate GitHub issue or pull request.
+- If a Maintainer of an affected git repository feels feedback from specific people is warranted they will @mention those users or teams to request feedback.
+- Proposals may also be added to the Flux Dev weekly meetings agenda, as a good avenue for making progress on a decision <https://lists.cncf.io/g/cncf-flux-dev/calendar>.
+- If there are objections and no consensus can be found, a vote may be called by a Maintainer.
+- When a vote is called, Maintainers will cast their yes/no vote on that GitHub issue or pull request, and after a suitable period of time, the outcome will be noted there.
```

Comment:
Is this any/all maintainers, or maintainers of a specific git repository? If we are going to have rules on voting, we should be clear on who gets to vote. (I would personally rather not have a voting process that includes all maintainers; it's too (easily) weighted towards a single vendor.)

Reply:
> Decisions that affect only one git repository may be made informally by its Maintainers.

> If there are objections and no consensus can be found, a vote may be called by a Maintainer.

> When a vote is called, Maintainers will cast their yes/no vote on that GitHub issue or pull request, and after a suitable period of time, the outcome will be noted there.

Together imply 

> maintainers of a specific git repository

but I do not think it would hurt to explicitly state this and make it:

> When a vote is called, Maintainers of the git repository will cast their yes/no vote on that GitHub issue or pull request, and after a suitable period of time, the outcome will be noted there.

Reply:
This is better. I think there's still some ambiguity over who gets a vote and who can reasonably say they have a stake. Without assuming malice, I think this could lead to votes being used to bang through decisions, because it's easier and faster than talking it through. So, I would rather have votes limited to the specific situations as set out (i.e., membership and governance). If this turns out to be impractical, we can reconsider.

Reply:
In the most recent commit, I proposed some very specific clarifications to address this concern.

Slight reorganization for clarity (into "Deciders" and "Kinds of Decisions"). And under Deciders, clarified when the deciders are a specific git repo's Maintainers or the Oversight Committee.

This is distinct from whether a decision can be lazy consensus by the deciders, should go through a proposal process for consensus by the deciders, or put to a vote by the deciders (or if the decision must be escalated).

WDYT?

Reply:
I ought not have understated my objection to voting, in the above.

The strongest kind of agreement is consensus; supermajorities and simple majorities are weaker, even if they feel more official.

From that perspective, only requiring a majority for maintainership, for example, seems rather poor. If you were a maintainer, and you felt strongly enough to go on record voting _against_ someone becoming a maintainer, would you be happy to lose the vote? That does not seem like it would be a good outcome for anyone.

My instinct is to see voting on technical decisions as divisive and ultimately corrosive. But I'd be prepared for individual projects to make their own mind up about that -- maybe the specific people involved are happy voting on low stakes decisions as a way to move fast. For decisions across several projects, I'd argue that voting for the sake of making it go fast is counter-productive.

Reply:
I can see maybe a tie-breaking vote might be required from time to time, but I would hope that it's used as a last resort and that everyone involved agrees a vote is needed (and if they do, they probably already know which way it's going to go -- so it's more a way of signalling that the decision is weaker than a consensus).

To approve this, I would want to see phrasing along the lines of:

 - most day-to-day decisions need only lazy consensus (PRs generally fall into this category, for example)
 - some specific kinds of "high stakes" decision require unanimity

Into the latter category I'd put:

 - election of new maintainers
 - everything currently under "supermajority"

An exception might be things which are high-stakes, but need to be enacted efficiently. Code of conduct violations and removing maintainers might fall here.

In summary: use a _stronger_ standard for governance-level decisions than for day to day decisions; emphasise consensus rather than voting.

Reply:
@squaremo I've tried to fold these concerns – and the chat with you and @hiddeco – into the most recent commit d52af2c. How is this looking?